compiler/rustc_interface/src/interface.rs

```diff
@@ -36,20 +36,11 @@ pub type Result<T> = result::Result<T, ErrorGuaranteed>;
 /// Can be used to run `rustc_interface` queries.
 /// Created by passing [`Config`] to [`run_compiler`].
 pub struct Compiler {
-    sess: Session,
-    codegen_backend: Box<dyn CodegenBackend>,
+    pub sess: Session,
+    pub codegen_backend: Box<dyn CodegenBackend>,
```

Comment:
The previous arrangement prevents modification in ways that would cause inconsistencies. I think we should keep read-only getters and have these fields private.

Reply:
But the only way to get a compiler is via `interface::run_compiler`, which gives you a `&Compiler`. So modification doesn't seem like a risk?

Reply:
Right